### DISCUSSION TITLE
Eeprom size compacting on axis

### DISCUSSION THREAD

**Discussion by manoukianv** [Ultrawipf/OpenFFBoard#37] [Ideas]
Hello,

i think, there is a litlle offset here in the memory pagination for the Effect1 address.
https://github.com/Ultrawipf/OpenFFBoard/blob/687796476e5d11101f57850762fa7e50e69fac63/Firmware/FFBoard/UserExtensions/Inc/eeprom_addresses.h#L80

There is 4 adresses reserved for the degree, but it seem to be only an uint16_t so required 1 adresses:
https://github.com/Ultrawipf/OpenFFBoard/blob/687796476e5d11101f57850762fa7e50e69fac63/Firmware/FFBoard/Src/NormalizedAxis.cpp#L68

hope it can help
Vincent

**Comment by Ultrawipf** [comment]
You might have misunderstood the addresses. They don't have to be continous values. Only one 16 bit field is used for the degrees. Just address 0x303.  There is a hole in the space to reserve some room for additional settings later. Does not have to be there but also does not hurt. They are just not used.

The eeprom emulation uses one 32 bit flash field by storing one 16 bit value together with its 16 bit address so any address is valid and fields that are never written also take up no space.

**Reply by manoukianv** [reply]
Thanks for explanation.
I saw the whole, but i don't think it's reserved space for later :-)
So i put the speed at the end of the Axis/NormalizedAxis eeprom segment, says me if you want i move it in the whole. I need 2*16b.
